File: examples/Node.js/dataStructures.ts
export interface PrivateApiKey {
  keyId: string;
  key: string;
  userId: string;
}

export type FormTemplate = Record<string, unknown>;

export interface NewFormSubmission {
  name: string;
}

export interface EncryptedFormSubmission {
  encryptedResponses: string;
  encryptedKey: string;
  encryptedNonce: string;
  encryptedAuthTag: string;
}

export interface FormSubmission {
  confirmationCode: string;
}

File: examples/.NET/DataStructures.cs
namespace dotnet
{
  public struct PrivateApiKey
  {
    public required string keyId { get; set; }
    public required string key { get; set; }
    public required string userId { get; set; }
  }

  public struct NewFormSubmission
  {
    public required string name { get; set; }
  }

  public struct EncryptedFormSubmission
  {
    public required string encryptedResponses { get; set; }
    public required string encryptedKey { get; set; }
    public required string encryptedNonce { get; set; }
    public required string encryptedAuthTag { get; set; }
  }

  public struct FormSubmission
  {
    public required string confirmationCode { get; set; }
  }
}